### PR TITLE
updates arith, data, lapack, blas documentation

### PR DESCRIPTION
**Pull request by melonakos**
Edited the <function-category>.(dox/h) Doxygen documentation prose. This does not include new examples.

Description
Better prose and function prototypes.

Changes to Users
None

Checklist
- [ ] Rebased on the latest master
- [x] Code compiles
- [x] Functions documented

### CODE REVIEW

File: include/af/arith.h [hunk 1]
```diff
@@ -75,7 +79,7 @@ namespace af
     /// \param[in] in input array
     /// \param[in] lo lower limit; can be an array or a scalar
     /// \param[in] hi upper limit; can be an array or a scalar
-    /// \return array containing values from \p in clamped between \p lo and \p hi
+    /// \return       clamped array
```
Comment (umar456) [inline include/af/arith.h:82]: I think the previous description was more informative.

File: include/af/arith.h [hunk 2]
```diff
@@ -322,79 +327,87 @@ namespace af
     /// @{
     /// C++ Interface to create a complex array from a single real array.
     ///
-    /// \param[in] in a real array
-    /// \return the returned complex array
+    /// \param[in] in input array
```
Comment (umar456) [inline include/af/arith.h:330]: I think this is less descriptive than the previous version. The user doesn't know it should be a real array from this parameter description.

File: include/af/arith.h [hunk 3]
```diff
@@ -322,79 +327,87 @@ namespace af
...
+    ///                  of the returned complex array
+    /// \return          complex array
     AFAPI array complex(const double real_, const array &imag_);
     /// @}
 
     /// C++ Interface to return the real part of a complex array.
     ///
     /// \param[in] in input complex array
-    /// \return real part
+    /// \return       real part
```
Comment (umar456) [inline include/af/arith.h:367]:
```suggestion
    /// \return    an array with the real values of the \p in array
```

File: include/af/arith.h [hunk 4]
```diff
@@ -419,19 +433,19 @@ namespace af
...
     ///
     AFAPI array pow2    (const array &in);
     /// @}
 
 #if AF_API_VERSION >= 31
     /// C++ Interface to evaluate the logistical sigmoid function.
     ///
-    /// \param[in] in input
-    /// \return sigmoid
+    /// Computes `1/(1+e^-x)`.
```
Comment (umar456) [inline include/af/arith.h:445]: LaTeX?

File: include/af/arith.h [hunk 5]
```diff
@@ -625,89 +645,106 @@ extern "C" {
...
+       elements of two arrays.
+
+       Output type is b8.
 
-       \param[out] out result of \p lhs < \p rhs; type is b8
-       \param[in] lhs first input
-       \param[in] rhs second input
-       \param[in] batch specifies if operations need to be performed in batch mode
-       \return \ref AF_SUCCESS if the execution completes properly
+       \param[out] out   <
```
Comment (umar456) [inline include/af/arith.h:664]: You shouldn't use a symbol to describe the parameter.

File: include/af/arith.h [hunk 6]
```diff
@@ -625,89 +645,106 @@ extern "C" {
...
+       elements of two arrays.
 
-       \param[out] out result of \p lhs > \p rhs; type is b8
-       \param[in] lhs first input
-       \param[in] rhs second input
-       \param[in] batch specifies if operations need to be performed in batch mode
-       \return \ref AF_SUCCESS if the execution completes properly
+       Output type is b8.
+
+       \param[out] out   >
```
Comment (umar456) [inline include/af/arith.h:680]: Same for the rest of the functions in this set.

File: include/af/blas.h [hunk 7]
```diff
@@ -7,165 +7,153 @@
...
-     */
+       \ingroup blas_func_matmul
+    */
     AFAPI array matmul(const array &lhs, const array &rhs,
                        const matProp optLhs = AF_MAT_NONE,
                        const matProp optRhs = AF_MAT_NONE);
 
     /**
-       \brief Matrix multiply of two arrays
+       C++ Interface to multiply two matrices.
```
Comment (umar456) [inline include/af/blas.h:49]: This function should discuss how the second parameter is transposed.

**Review by umar456 — CHANGES_REQUESTED** [review]

File: docs/details/arith.dox [hunk 8]
```diff
@@ -1,6 +1,6 @@
 /*!
 \page arith_real_only arith_real
-\note This function supports real inputs only. Complex inputs are not yet supported.
+This function only supports real inputs; complex inputs are not yet supported.
```
[the remark this answers is not in the document]
Comment (melonakos) [reply docs/details/arith.dox:3]: fixed

File: docs/details/blas.dox [hunk 9]
```diff
@@ -1,29 +1,18 @@
...
-Scalar dot product between two vectors.  Also referred to as the inner
-product.
-
-=======================================================================
 
 \defgroup blas_func_matmul matmul
-\ingroup blas_mat
 
-\brief Matrix multiplication using array
+Multiply two matrices.
```
[the remark this answers is not in the document]
Comment (melonakos) [reply docs/details/blas.dox:7]: fixed to "matrix multiplication." I don't use the word arrays in the brief description. all things work on arrays. it is implied.

File: docs/details/data.dox [hunk 10]
```diff
@@ -87,7 +78,8 @@ array b = iota(dim4(5, 3), dim4(1, 2))
 =======================================================================
 
 \defgroup data_func_diag diag
-\brief Extract diagonal from a matrix when \p extract is set to true. Create a diagonal matrix from input array when \p extract is set to false
+
+Extract the diagonal from an array. Optionally, form a diagonal matrix.
```
[the remark this answers is not in the document]
Comment (melonakos) [reply docs/details/data.dox:82]: done

File: docs/details/data.dox [hunk 11]
```diff
@@ -151,13 +144,13 @@ Requires that all dimensions except the join dimension must be the same for all
 
 \defgroup manip_func_tile tile
 
-\brief Repeat the contents of the input array along the specified dimensions
+Generate a tiled array.
```
[the remark this answers is not in the document]
Comment (melonakos) [reply docs/details/data.dox:147]: done

File: docs/details/data.dox [hunk 12]
```diff
@@ -184,7 +177,7 @@ dimension:
 
 \defgroup manip_func_reorder reorder
 
-\brief Reorder an array according to the specified dimensions.
+Reorder an array.
```
[the remark this answers is not in the document]
Comment (melonakos) [reply docs/details/data.dox:180]: this is a brief description. the immediately following more verbose description describes what you are looking for.

File: include/af/arith.h (hunk 3 above)
Comment (melonakos) [reply include/af/arith.h:367]: you need to look at the HTML to see that this is descriptive enough already.

File: include/af/arith.h (hunk 4 above)
Comment (melonakos) [reply include/af/arith.h:445]: done

File: include/af/arith.h (hunk 5 above)
Comment (melonakos) [reply include/af/arith.h:664]: i think it was fine, but i changed it.

File: include/af/arith.h (hunk 6 above)
Comment (melonakos) [reply include/af/arith.h:680]: i think it was fine, but i changed it.

File: include/af/blas.h (hunk 7 above)
Comment (melonakos) [reply include/af/blas.h:49]: Not sure what you mean. The optLhs and optRhs parameters are described.

**Review by syurkevi — APPROVED** [review]

**Pull request merged**